src/components/DetailsView/index.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useContext } from 'react';
+import ModiteContext from '../../state/modite';
+import Modite from '../../models/Modite';
+import classNames from 'classnames/bind';
+import s from './styles.module.css';
+import ModiteProfileResp from '../../models/ModiteProfileResp';
+
+// TODO: type correctly
+function Details({ className = '' } : any) {
+  const [activeModite, setActiveModite]: [Modite, React.Dispatch<any>] = useContext(ModiteContext);
+  const { profile = {} }: any = activeModite || {};
+  let { fields } = profile;
+
+  const fetchProfile = async () => {
+    if (!activeModite) return;
+
+    const moditeProfile: ModiteProfileResp = await fetch(
+      `https://modus.app/modite/${activeModite.id}`,
+    ).then(res => res.json());
+    activeModite.profile = moditeProfile.profile;
+    fields = moditeProfile.profile.fields;
+  }
+
+  if (!fields) {
+    fetchProfile();
+    fields = {};
+  }
+
+  const image = profile && profile.image_192;
+  const name = activeModite && activeModite.real_name ? activeModite && activeModite.real_name : activeModite && activeModite.name;
+  const { Location: location } = fields;
+  const tod = activeModite && activeModite.tod;
+  const localDate = activeModite && activeModite.localDate;
+  const localTime = activeModite && activeModite.localTime;
+
+  const cx = classNames.bind(s);
+  className = cx('moditeCt', className);
+
+  return (
```

Comment:
For MVP I'm fine with having this here even if there is no `activeModite` so it can be animated up but feels like there is a better solution here. Like return `null` if no `activeModite` so nothing is rendered but when one is present render off screen and animate up. This would also help with the const block where you have to check if `activeModite` is truthy many times.

Reply:
I think you're right.  Maybe we don't fix it now, but the checks are piling up and there's got to be a better way to account for presence.  I forgot you could just return null on a component in React, but that probably is the most elegant solution here.